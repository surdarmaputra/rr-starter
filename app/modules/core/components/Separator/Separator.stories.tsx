import type { Meta, StoryObj } from '@storybook/react';

import { Separator, type SeparatorProps } from './Separator';

type Story = StoryObj<typeof Separator>;

const meta: Meta = {
  component: Separator,
  title: 'Core/Components/Separator',
};

export default meta;

export const Default: Story = {
  render: (args: SeparatorProps) => (
    <div className="flex flex-col gap-2">
      Sample Text
      <Separator {...args} />
      Sample Text
      <Separator />
      <div className="flex h-10 gap-2">
        Text
        <Separator orientation="vertical" />
        Text
      </div>
    </div>
  ),
};
