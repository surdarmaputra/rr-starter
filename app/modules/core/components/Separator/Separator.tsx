import * as SeparatorPrimitive from '@radix-ui/react-separator';
import * as React from 'react';

import { mergeClassNames } from '~/modules/core/libs/utils/ui';

export type SeparatorProps = React.ComponentPropsWithoutRef<
  typeof SeparatorPrimitive.Root
>;

const Separator = React.forwardRef<
  React.ComponentRef<typeof SeparatorPrimitive.Root>,
  SeparatorProps
>(
  (
    { className, orientation = 'horizontal', decorative = true, ...props },
    ref,
  ) => (
    <SeparatorPrimitive.Root
      className={mergeClassNames(
        'shrink-0 bg-divider',
        orientation === 'horizontal' ? 'h-[1px] w-full' : 'h-full w-[1px]',
        className,
      )}
      decorative={decorative}
      orientation={orientation}
      ref={ref}
      {...props}
    />
  ),
);
Separator.displayName = SeparatorPrimitive.Root.displayName;

export { Separator };
