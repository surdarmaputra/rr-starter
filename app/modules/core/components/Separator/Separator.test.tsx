import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { Separator } from './Separator';

describe('Separator', () => {
  function renderComponent() {
    return render(<Separator data-testid="Separator-testid" />);
  }

  test('renders with data-testid', async () => {
    renderComponent();
    await screen.findByTestId('Separator-testid');
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
