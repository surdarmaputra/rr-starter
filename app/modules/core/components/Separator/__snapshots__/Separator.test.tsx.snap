// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Separator > renders snapshot 1`] = `
<div>
  <div
    class="shrink-0 bg-divider h-[1px] w-full"
    data-orientation="horizontal"
    data-testid="Separator-testid"
    role="none"
  />
</div>
`;
